File: problem-solving/baekjoon/10773_제로.py
import sys

class Stack:
    def __init__(self):
        self.arr = []
        self.top = -1
      
    def get_size(self):
        return self.top + 1
    
    def is_empty(self):
        if self.top == -1:
            return True
        return False
        
    def push(self, n):
        self.top += 1
        self.arr.append(n)

    def pop(self):
        if self.is_empty():
            return -1
        result = self.arr[self.top]
        del self.arr[self.top]
        self.top -= 1
        return result

    def get_top(self):
        if self.is_empty():
            return -1
        return self.arr[self.top]

if __name__ == "__main__":
    k = int(sys.stdin.readline())
    stack = Stack()
    for i in range(0,k):
        num = int(sys.stdin.readline())
        if num == 0:
            stack.pop()
            continue
        stack.push(num)
    sum = 0
    while True:
        val = stack.pop()
        if val == -1:
            break
        sum += val
    print(sum)

File: problem-solving/baekjoon/10773_제로.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)

	var k int
	fmt.Fscanln(reader, &k)

	var s = NewStack()
	s.Init()
	for i := 0; i < k; i++ {
		var num int
		fmt.Fscanln(reader, &num)
		if num == 0 {
			s.Pop()
			continue
		}
		s.Push(num)
	}

	var sum int
	for true {
		val := s.Pop()
		if val == -1 {
			break
		}
		sum += val
	}
	fmt.Println(sum)
}

type Stack interface {
	GetSize() int
	IsEmpty() bool
	Push(int)
	Pop() int
	GetTop() int
	Init()
}

type stack struct {
	arr [100000]int
	top int
}

func NewStack() Stack {
	return &stack{}
}

func (s *stack) Init() {
	s.top = -1
}

func (s *stack) GetSize() int {
	return s.top + 1
}
func (s *stack) IsEmpty() bool {
	if s.top == -1 {
		return true
	}
	return false
}

func (s *stack) Push(n int) {
	s.top++
	s.arr[s.top] = n
}

func (s *stack) Pop() int {
	if s.IsEmpty() {
		return -1
	}
	result := s.arr[s.top]
	s.top--
	return result
}

func (s *stack) GetTop() int {
	if s.IsEmpty() {
		return -1
	}
	return s.arr[s.top]
}
